src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+  }
+
+  #handlePrev = () => {
+    if (this.actualPage > 1) {
+      this.actualPage -= 1;
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleNext = () => {
+    if (
+      this.actualPage < this.totalPages ||
+      (this.totalPages < 0 && this.hasNextPage)
+    ) {
+      this.actualPage += 1;
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleFirst = () => {
+    if (this.actualPage !== 1) {
+      this.actualPage = 1;
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleLast = () => {
+    if (
+      this.actualPage !== this.totalPages ||
+      (this.totalPages < 0 && this.hasNextPage)
+    ) {
+      this.actualPage = this.totalPages;
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleItemsPerPageChange = (event: Event) => {
+    const select = event.target as HTMLSelectElement;
+    const newItemsPerPage = Number(select.value);
+    this.itemsPerPage = newItemsPerPage;
+    this.itemsPerPageChange.emit(newItemsPerPage);
+  };
+
+  #handleGoToOnKeyDown = (event: KeyboardEvent) => {
+    if (event.key === "Enter") {
+      this.#handleGotoPageChange(event);
+    }
+  };
+
+  #handleGoToOnBlur = (event: FocusEvent) => {
+    this.#handleGotoPageChange(event);
+  };
+
+  #handleGotoPageChange = (event: Event) => {
+    const input = event.target as HTMLInputElement;
+
+    const page = Number(input.value);
+
+    if (this.totalPages < 0) {
+      this.pageChange.emit(page);
+    }
+
+    // Validate if page exceeds total pages range
+    if (page >= 1 && page <= this.totalPages) {
+      this.actualPage = Math.max(Math.min(page, this.totalPages), 1);
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleSelectedPageChange = (event: PointerEvent) => {
+    event.stopPropagation();
+    const elementRef = event.composedPath()[0] as
+      | HTMLAnchorElement
+      | HTMLButtonElement;
+
+    // Check the click event is performed in a button and anchor element
+    if (
+      elementRef.tagName.toLowerCase() !== "button" &&
+      elementRef.tagName.toLowerCase() !== "a"
+    ) {
+      return;
+    }
+    event.preventDefault();
+
+    let newPageSelected;
+
+    if (elementRef.tagName.toLowerCase() === "a") {
+      newPageSelected = this.hasHyperlink
+        ? (elementRef as HTMLAnchorElement)?.href
+        : elementRef?.id;
+    } else {
+      newPageSelected = (elementRef as HTMLButtonElement)?.value;
+    }
+
+    // Don't fire the selectedItemChange event if the button is already selected
+    if (this.selectedPage === newPageSelected) {
+      return;
+    }
+    this.selectedPage = newPageSelected;
+
+    const eventInfo = this.pageChange.emit(this.actualPage);
+
+    if (!eventInfo.defaultPrevented) {
+      this.selectedPage = newPageSelected;
+    }
+  };
+
+  #renderItemsPerPage = () => {
+    return (
+      <form
+        class="form"
+        part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE]: true })}
```

Comment:
If all parts are `true`, we can avoid using the tokenMap to improve the rendering performance.

```suggestion
        part={PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE}
```